HomeUI/src/views/apps/Management.vue

```diff
@@ -6979,50 +7178,764 @@ export default {
     this.getMultiplier();
     this.getEnterpriseNodes();
     this.getDaemonBlockCount();
+    // this.initCharts();
   },
   beforeDestroy() {
     this.stopPolling();
+    this.stopPollingStats();
     window.removeEventListener('resize', this.onResize);
   },
   methods: {
-    extractTimestamp(log) {
-      return log.split(' ')[0];
-    },
-    toggleLogSelection(log) {
-      const logTimestamp = this.extractTimestamp(log);
-      if (this.selectedLog.includes(logTimestamp)) {
-        this.selectedLog = this.selectedLog.filter((ts) => ts !== logTimestamp);
+    // Stats Section START
+    enableHistoryStatisticsChange() {
+      this.buttonStats = false;
+      this.noData = false;
+      if (this.enableHistoryStatistics) {
+        this.stopPollingStats();
+        this.clearCharts();
+        this.fetchStats();
       } else {
-        this.selectedLog.push(logTimestamp);
+        this.clearCharts();
+        this.startPollingStats();
       }
     },
-    unselectText() {
-      this.selectedLog = [];
+    LimitChartItems(chart) {
+      const datasetLength = chart.data.datasets[0].data.length;
+      if (datasetLength > this.selectedPoints) {
+        const excess = datasetLength - this.selectedPoints;
+        chart.data.labels = chart.data.labels.slice(excess);
+        chart.data.datasets.forEach((dataset) => {
+          dataset.data = dataset.data.slice(excess);
+        });
+        chart.update({
+          duration: 800,
+          lazy: false,
+          easing: 'easeOutBounce',
+        });
+      }
     },
-    async copyCode() {
+    async scrollToPagination() {
+      await this.$nextTick();
+      window.scrollTo(0, document.body.scrollHeight);
+    },
+    getHddByName(applications, appName) {
+      if (applications?.compose) {
+        const app = applications.compose.find((application) => application.name === appName);
+        return app.hdd;
+      // eslint-disable-next-line no-else-return
+      } else {
+        return applications.hdd;
+      }
+    },
+    processStatsData(statsData, configData, timeStamp = null) {
+      const memoryLimitBytes = statsData.memory_stats.limit;
+      this.memoryLimit = memoryLimitBytes;
+      const memoryUsageBytes = statsData.memory_stats?.usage || null;
+      const memoryUsageMB = memoryUsageBytes;
+      const memoryUsagePercentage = ((memoryUsageBytes / memoryLimitBytes) * 100).toFixed(1);
+      const cpuUsage = statsData.cpu_stats.cpu_usage.total_usage - statsData.precpu_stats.cpu_usage.total_usage;
+      const systemCpuUsage = statsData.cpu_stats.system_cpu_usage - statsData.precpu_stats.system_cpu_usage;
+      const onlineCpus = statsData.cpu_stats.online_cpus;
+      const nanoCpus = configData.HostConfig.NanoCpus;
+      const cpuSize = (((cpuUsage / systemCpuUsage) * onlineCpus)).toFixed(1) || 0;
+      // eslint-disable-next-line no-mixed-operators
+      const cpuPercent = (cpuSize / (nanoCpus / 1e9) * 100).toFixed(1);
```

Comment:
I don't see in the new method the app specs cpu being used to do the calculation like we had on previous
const cpu = `${(((cpuUsage / systemCpuUsage) * entry.data.cpu_stats.online_cpus * 100) / (nanoCpus / specifications.cpu / 1e9) || 0).toFixed(2)}%`;
we are using different formula, why?